src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java

```diff
@@ -71,7 +71,8 @@ public boolean start() throws Exception {
         node.addAction(new PauseAction("Input"));
 
         String baseUrl = '/' + run.getUrl() + getPauseAction().getUrlName() + '/';
-        if (input.getParameters().isEmpty()) {
+        //JENKINS-40594 submitterParameter does not work without at least one actual parameter
+        if (input.getParameters().isEmpty() && (input.getSubmitterParameter() == null || input.getSubmitterParameter().isEmpty())) {
```

Comment:
Due to [this trimming](https://github.com/jenkinsci/pipeline-input-step-plugin/blob/ab59e21435764953138782a0abdb2e13f989d28c/src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStep.java#L84) it should suffice to say

```java
if (input.getParameters().isEmpty() && input.getSubmitterParameter() == null) {
```